### PR TITLE
added middleware to /fork endpoint to check if project can be forked,…

### PR DESCRIPTION
**Pull request by joefitter**
… if not return the latest editable version

### CODE REVIEW

File: lib/routers/establishment/projects.js [hunk 1]
```diff
@@ -97,27 +100,42 @@ router.param('id', (req, res, next, id) => {
...
     .catch(next);
 });
 
+const canFork = (req, res, next) => {
+  const { ProjectVersion } = req.models;
+  ProjectVersion.query()
+    .select('id', 'grantedAt', 'submittedAt', 'createdAt')
+    .where({ projectId: req.project.id })
+    .orderBy('createdAt', 'desc')
+    .then(versions => versions[0])
```
Comment (lennym) [inline lib/routers/establishment/projects.js]: Are the versions not already available on `req.project.versions`? Is there a need to go back to the database here?

Comment (joefitter) [reply lib/routers/establishment/projects.js]: good spot 👍

File: lib/routers/establishment/projects.js [hunk 2]
```diff
@@ -97,27 +103,35 @@ router.param('id', (req, res, next, id) => {
...
 
+const canFork = (req, res, next) => {
+  const version = get(req.project, 'versions[0]');
+  if (!version) {
+    return next(new NotFoundError());
+  }
+  // version can be forked, continue
+  if (version.grantedAt || version.submittedAt) {
+    return next();
+  }
```
Comment (lennym) [inline lib/routers/establishment/projects.js]: Is this now incompatible with https://github.com/UKHomeOffice/asl-schema/pull/138 ?

Comment (joefitter) [reply lib/routers/establishment/projects.js]: yes

**Pull request merged**